pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -522,6 +522,7 @@ func TestDesiredRouterDeployment(t *testing.T) {
 	checkDeploymentHasEnvVar(t, deployment, "ROUTER_UNIQUE_ID_FORMAT", true, `"foo"`)
 
 	checkDeploymentHasEnvVar(t, deployment, "ROUTER_H1_CASE_ADJUST", false, "")
+	checkDeploymentHasEnvVar(t, deployment, RouterHardStopAfterEnvName, false, "")
```

Comment:
We should add test coverage somewhere in this unit test for when the annotation is set.

Reply:
> We should add test coverage somewhere in this unit test for when the annotation is set.

In progress.